src/execute_tracked_trajectory.cpp

```diff
@@ -0,0 +1,228 @@
+  }
+  goal_handle_ = future_goal_handle.get();
+  if (!goal_handle_)
+  {
+    trigger_service_.reset();
+    return tl::make_unexpected("Goal request was rejected by the server.");
+  }
+
+  // Wait for result (with optional timeout)
+  auto future_result = action_client_->async_get_result(goal_handle_);
+  if (duration_tolerance >= 0.0)
+  {
+    if (future_result.wait_for(std::chrono::duration<double>(duration_tolerance)) == std::future_status::timeout)
+    {
+      action_client_->async_cancel_goal(goal_handle_);
+      trigger_service_.reset();
+      return tl::make_unexpected("Timed out waiting for result.");
+    }
+  }
+  auto wrapped_result = future_result.get();
+
+  // Handle completion or cancellation
+  switch (wrapped_result.code)
+  {
+    case rclcpp_action::ResultCode::SUCCEEDED:
+      trigger_service_.reset();
+      return true;
+
+    case rclcpp_action::ResultCode::CANCELED:
+      trigger_service_.reset();
+      if (triggered_)
+      {
+        auto rem = computeRemainderTrajectory();
+        setOutput<trajectory_msgs::msg::JointTrajectory>("trajectory_remainder", rem);
+        return tl::make_unexpected("Trajectory halted");
+      }
+      else
+      {
+        return tl::make_unexpected("Trajectory canceled unexpectedly");
+      }
+
+    case rclcpp_action::ResultCode::ABORTED:
+      trigger_service_.reset();
+      return tl::make_unexpected("Action aborted: " + wrapped_result.result->error_string);
+
+    default:
+      trigger_service_.reset();
+      return tl::make_unexpected("Unknown action result code");
+  }
+}
+
+tl::expected<void, std::string> ExecuteAndTrackJointTrajectory::doHalt()
+{
+  triggered_ = true;
+  if (goal_handle_)
+  {
+    try
+    {
+      action_client_->async_cancel_goal(goal_handle_);
+    }
+    catch (const rclcpp_action::exceptions::UnknownGoalHandleError& e)
+    {
+      RCLCPP_WARN(rclcpp::get_logger("ExecuteAndTrackJointTrajectory"), "doHalt cancel failed: %s", e.what());
+    }
+    goal_handle_.reset();
+  }
+  trigger_service_.reset();
+  return {};
+}
+
+void ExecuteAndTrackJointTrajectory::handleTrigger(const std::shared_ptr<std_srvs::srv::Trigger::Response> res)
+{
+  triggered_ = true;
+  if (goal_handle_)
+  {
+    try
+    {
+      action_client_->async_cancel_goal(goal_handle_);
+    }
+    catch (const rclcpp_action::exceptions::UnknownGoalHandleError& e)
+    {
+      RCLCPP_WARN(rclcpp::get_logger("ExecuteAndTrackJointTrajectory"), "Failed to cancel goal: %s", e.what());
+    }
+    goal_handle_.reset();
+  }
+  res->success = true;
+  res->message = "Trajectory cancellation triggered";
+}
+
+void ExecuteAndTrackJointTrajectory::processFeedback(
+    const std::shared_ptr<const FollowJointTrajectory::Feedback> feedback)
+{
+  last_feedback_time_ = feedback->desired.time_from_start.sec + feedback->desired.time_from_start.nanosec * 1e-9;
+  setOutput<double>("time_from_start", last_feedback_time_);
```

Comment:
Do we have to set `time_from_start` every time the feedback is processed or only when the trajectory is paused?